configreader/reader_test.go

```diff
@@ -0,0 +1,194 @@
+							{
+
+								Match: &envoy_config_route_v3.RouteMatch{
+									PathSpecifier: &envoy_config_route_v3.RouteMatch_Prefix{
+										Prefix: "/",
+									},
+								},
+								Action: &envoy_config_route_v3.Route_Route{
+									Route: &envoy_config_route_v3.RouteAction{
+										ClusterSpecifier: &envoy_config_route_v3.RouteAction_Cluster{
+											Cluster: "some_service",
+										},
+									},
+								},
+							},
+						},
+					},
+				},
+			},
+		},
+		HttpFilters: []*http_manger_v3.HttpFilter{
+			{Name: "envoy.filters.http.router"},
+		},
+	})
+	staticBootstrap = &v3.Bootstrap{
+		Admin: &v3.Admin{
+			AccessLogPath: "/tmp/admin_access.log",
+			Address: &envoy_config_core_v3.Address{
+				Address: &envoy_config_core_v3.Address_SocketAddress{
+					SocketAddress: &envoy_config_core_v3.SocketAddress{
+						Address: "127.0.0.1",
+						PortSpecifier: &envoy_config_core_v3.SocketAddress_PortValue{
+							PortValue: 9901,
+						},
+					},
+				},
+			},
+		},
+		StaticResources: &v3.Bootstrap_StaticResources{
+			Clusters: []*v36.Cluster{
+				{
+					Name: "some_service",
+					ClusterDiscoveryType: &v36.Cluster_Type{
+						Type: v36.Cluster_STATIC,
+					},
+					ConnectTimeout: &duration.Duration{Nanos: 250000000},
+					LoadAssignment: &v31.ClusterLoadAssignment{
+						ClusterName: "some_service",
+						Endpoints: []*v31.LocalityLbEndpoints{
+							{
+								LbEndpoints: []*v31.LbEndpoint{
+									{
+										HostIdentifier: &envoy_config_endpoint_v3.LbEndpoint_Endpoint{
+											Endpoint: &envoy_config_endpoint_v3.Endpoint{
+												Address: &envoy_config_core_v3.Address{
+													Address: &envoy_config_core_v3.Address_SocketAddress{
+														SocketAddress: &envoy_config_core_v3.SocketAddress{
+															Address: "127.0.0.1",
+															PortSpecifier: &envoy_config_core_v3.SocketAddress_PortValue{
+																PortValue: 1234,
+															},
+														},
+													},
+												},
+											},
+										},
+									},
+								},
+							},
+						},
+					},
+				},
+			},
+			Listeners: []*v35.Listener{
+				{
+					Name: "listener_0",
+					Address: &envoy_config_core_v3.Address{
+						Address: &envoy_config_core_v3.Address_SocketAddress{
+							SocketAddress: &envoy_config_core_v3.SocketAddress{
+								Address: "127.0.0.1",
+								PortSpecifier: &envoy_config_core_v3.SocketAddress_PortValue{
+									PortValue: 10000,
+								},
+							},
+						},
+					},
+					FilterChains: []*envoy_config_listener_v3.FilterChain{
+						{
+							Filters: []*envoy_config_listener_v3.Filter{
+								{
+									Name: "envoy.filters.network.http_connection_manager",
+									ConfigType: &v35.Filter_TypedConfig{
+										TypedConfig: httpConnectionManager,
+									},
+								},
+							},
+						},
+					},
+				},
+			},
+		},
+	}
+	if err != nil {
+		panic(err)
+	}
+}
+
+func TestReader(t *testing.T) {
+	type readerTest struct {
```

Comment:
This is a weird way to write a table test. I usually see:

```go
tcs := []struct{
  name string
  input string
  expected *v3.Bootstrap
}{
  {name: "basic", input: "../testdata/static.yaml", expected: staticBootstrap}
  {name: "less basic", input: "../testdata/dynamic.yaml", expected: dynamicBootstrap}

}

for _, tc := range tcs {
  t.Run(tc.name, func(t *testing.T) {
  ...
  })
}
```

Reply:
I also don't know if you need a table test for a single case. It might all read more easily in a single func instead of spread out between the test, a `var`, and `init`.

Reply:
The idea of the table test was to motivate me (or others) to add more in the future